pkg/chrootarchive/archive.go

```diff
@@ -48,16 +46,20 @@ func untarHandler(tarArchive io.Reader, dest string, options *archive.TarOptions
 		}
 	}
 
-	r := ioutil.NopCloser(tarArchive)
+	r := tarArchive
```

Comment:
that is to fix the 2nd race I mentioned in [comment](https://github.com/docker/docker/issues/16924#issuecomment-147250089) on io.File, The root cause is
pkg/archive.(*Archiver).UntarPath() has a defer'ed call to archive.Close(), while exec.Cmd has a goroutine do io.Copy at the same time; although that race didn't cause crash problem.

this change to get rid of ioutil.NopCloser is to move responsiblity to untarHandler to determine a later proper time (or in another goroutine) to call that .Close
https://github.com/docker/docker/blob/master/pkg/archive/archive.go#L785
https://github.com/c0b/docker/blob/fix-race-16924/pkg/archive/archive.go#L803-L804

I agree there are more untarHandler needs to be changed, I only changed the one in chrootarchive/archive.go fixed it on Linux.